### PR TITLE
Update roadmap

### PR DESCRIPTION
**Pull request by nerdneha**
[Rendered Markdown](https://github.com/desktop/desktop/blob/long-and-winding-roadmap/docs/process/roadmap.md)

includes releases and very rough sketch of features for 1.2+
* includes rough idea of releases
* will change based on reality
* we will re-evaluate at the 6 month mark in August
* subject to change as new features are released on GitHub.com or as we discover better opportunities to impact the user positively

way of the future

### CODE REVIEW

File: docs/process/roadmap.md [hunk 1]
```diff
@@ -1,63 +1,70 @@
...
+- [Upgrade nudge from classic to the new app](https://github.com/desktop/desktop/issues/2146)
 
-- Conflict resolution lite
-  - Differentiate conflicted files from other changed files
-  - Let me choose my mergetool
-  - Let me open my mergetool
+## 1.3 - Detecting marge conflicts - 2.0
+  
+- [Relationship between branches 2.0](https://github.com/desktop/desktop/issues/2639)
+  - Notifications around diverging from master branch
```
Comment (donokuda) [inline docs/process/roadmap.md]: I think @niik's [comment here](https://github.com/desktop/desktop/issues/2639#issuecomment-340791736) explains the notifications part a bit. Should we consider linking this bullet point to there?

File: docs/process/roadmap.md [hunk 2]
```diff
@@ -1,63 +1,70 @@
...
+  - Compare my work to any branch
+  - Merge in any other branch to my branch
+  
+- [Upgrade nudge from classic to the new app](https://github.com/desktop/desktop/issues/2146)
 
-- Conflict resolution lite
-  - Differentiate conflicted files from other changed files
-  - Let me choose my mergetool
-  - Let me open my mergetool
+## 1.3 - Detecting marge conflicts - 2.0
```
Comment (donokuda) [inline docs/process/roadmap.md]: > marge conflicts

image

File: docs/process/roadmap.md [hunk 3]
```diff
@@ -1,63 +1,70 @@
...
+- Better onboarding for new users
 
-- Auto-config upstreams on fork
-  - Create an `upstream` remote for forks
-  - Fetch periodically
+## 1.6 - Conflict resolution lite
 
-## 1.2
+- [Conflict resolution lite](https://github.com/desktop/desktop/issues/2640)
+  - Differentiate conflicted files from other changed files
```
Comment (donokuda) [inline docs/process/roadmap.md]: "Differentiate" seems to be a little confusing based on some of the comments in that issue. Maybe something like "Separate conflicted files from other changed files"?

File: docs/process/roadmap.md [hunk 4]
```diff
@@ -1,63 +1,70 @@
...
-- Protected branches and default branch protection
-  - Don't let me commit to branches I shouldn't commit to
-  - Guide me to creating my own branches
+- App interaction tracking
 
-- CI status notifications lite
-  - Only on my PRs
-  - Use OS notifications
-  - Notifications link to the PR on dotcom
+- Human readable Table of Contents
```
Comment (donokuda) [inline docs/process/roadmap.md]: Maybe something like "View changes by methods and functions (for supported files)?"

I'm assuming we were talking implementing something like [this](https://blog.github.com/2017-07-26-quickly-review-changed-methods-and-functions-in-your-pull-requests/)?

File: docs/process/roadmap.md [hunk 5]
```diff
@@ -1,63 +1,70 @@
...
+  
+- End of Progress & in-app messaging
 
-- Fork if needed
-  - Offer to fork on clone, or push, or ?
-
-- Protected branches and default branch protection
-  - Don't let me commit to branches I shouldn't commit to
-  - Guide me to creating my own branches
+- App interaction tracking
```
Comment (donokuda) [inline docs/process/roadmap.md]: > End of progress & in-app messaging
>
> App interaction tracking

I.. totally forgot what these were. Can any shed some clarity for me? 😅

File: docs/process/roadmap.md (hunk 2 above)
Comment (shiftkey) [reply docs/process/roadmap.md]: Can we just leave this as-is? :trollface:

Comment (shiftkey) [reply docs/process/roadmap.md]: But more seriously, what's the `2.0` part about this represent? Conflict resolution **lite** is set down for `1.6` - should this be `1.0` instead?

File: docs/process/roadmap.md [hunk 6]
```diff
@@ -1,63 +1,71 @@
...
 
-- CI status notifications lite
-  - Only on my PRs
-  - Use OS notifications
-  - Notifications link to the PR on dotcom
+- Human readable Table of Contents
 
-## 1.3
+## Icebox
```
Comment (shiftkey) [inline docs/process/roadmap.md]: This is totally a leading question, and overlaps with what I've been pondering about, but does this stuff belong on the roadmap? Or can it live elsewhere?

File: docs/process/roadmap.md (hunk 5 above)
Comment (shiftkey) [reply docs/process/roadmap.md]: @donokuda attempting to remember our earlier discussions:

 - "End of Progress" was calling out some of the flows where we don't make it clear that they're done - we just wrap it up and return the user to the main view. The classic example of this is login - it's not clear that your login succeeded, the flow just finishes.

 - I'm a bit hazy on "in-app messaging" and can't remember the concrete things we talked about

 - "App interaction tracking" - as we roll out features, it'd be great to understand just how much they're being used in the wild. We've been haphazard about this in the past, so this was a chance for us to revisit this.

File: docs/process/roadmap.md [hunk 7]
```diff
@@ -1,63 +1,71 @@
...
+The following are our marketing releases, as planned to date (and we plan to revisit 6 months from now). They contain the broad features we're considering. The further away a release is, the less defined it is. Releases are subject to change at any time. This is not written in stone!!!
 
-## 1.1
+## 1.2 - Relationship between branches - lite
 
-- Relationship to _X_
-  - Show how the current branch relates to the target branch
-  - The target is the default branch or the upstream's default branch
-  - _Probably_ let the user switch the target branch
+- [Relationship between branches lite](https://github.com/desktop/desktop/issues/3956) - [#3956](https://github.com/desktop/desktop/issues/3956)
```
Comment (shiftkey) [inline docs/process/roadmap.md]: We don't explain what "lite", "1.0" and "2.0" represent in terms of this document. I took a stab at this here, see if you like it or not:

> A feature may be suffixed with a label like "lite" or "1.0" or "2.0" - this is used to define the scope of the feature, as it is expected to evolve over time:
>
> - "lite" - the fundamentals of the feature are implemented, enough to get the common tasks done
> - "1.0" - the feature has evolved beyond the fundamentals, enough for every day use
> - "2.0" - with more knowledge about the domain and use cases, the feature now supports advanced tasks

File: docs/process/roadmap.md (hunk 5 above)
Comment (donokuda) [reply docs/process/roadmap.md]: Crystal clear now 💎thanks, @shiftkey!

**Comment by nerdneha** [comment]
@shiftkey @donokuda 

Not sure if this was too 🔥 ✊ but I pushed up a version without `lite` and `2.0` to see if it was possible. Instead, I used `MVP` and `continued`. Is this still confusing? Is it worse? Would love to hear your thoughts. 

also thanks for the rest of the feedback! removed the icebox and added some clarity around later releases. more than happy to go in and add even more detail!

**Comment by nerdneha** [comment]
🦑

**Comment by niik** [comment]
Gonna give this to @donokuda to merge just to verify that he's happy with the changes.

**Comment by niik** [comment]
@nerdneha I'm assuming this was `ready-for-review`?

**Comment by shiftkey** [comment]
>  instead, I used MVP and continued. is this still confusing? Is it worse? would love to hear your thoughts.

@nerdneha 👍 yeah, this works

**Comment by donokuda** [comment]
giphy 4

**Pull request merged**